Code/Editor/CryEdit.cpp

```diff
@@ -2299,9 +2299,25 @@ int CCryEditApp::IdleProcessing(bool bBackgroundUpdate)
             GetIEditor()->Notify(eNotify_OnIdleUpdate);
         }
     }
-    else if (GetIEditor()->GetSystem() && GetIEditor()->GetSystem()->GetILog())
+    else
     {
-        GetIEditor()->GetSystem()->GetILog()->Update(); // print messages from other threads
+        if (GetIEditor()->GetSystem() && GetIEditor()->GetSystem()->GetILog())
+        {
+            GetIEditor()->GetSystem()->GetILog()->Update(); // print messages from other threads
+        }
+
+        // If we're backgrounded and not fully background updating, idle to rate limit SystemTick
+        const float safeMarginFPS = 0.5f; // safe margin to not drop below idle fps
+        const int maxEditorFPS = gEnv->pConsole->GetCVar("ed_backgroundSystemTickFPS")->GetIVal();
+        static AZ::TimeMs sTimeLast = AZ::GetRealElapsedTimeMs();
+        const AZ::TimeMs timeFrameMax(static_cast<AZ::TimeMs>((int64)(1000.f / ((float)maxEditorFPS + safeMarginFPS))));
```

Comment:
Nit: the naming of the variables here is a bit confusing. This time here isn't so much `timeFrameMax` but is instead the _minimum_ time that is allowed to elapse on a given frame in the backgrounded situation correct?

Would suggest something like:
```
AZ::TimeMs minFrameTime = ...
AZ::TimeMs elapsedFrameTime = AZ::GetRealElapsedTimeMs() - sTimeLast;
if (elapsedFrameTime < minFrameTime)
{
    // sleep
}
sTimeLast = AZ::GetRealElapsedTimeMs();
```

Reply:
It depends on how you look at it. I read the original version as "maximum amount of ms a frame can take without going over budget". That being said, the naming could stand a cleanup pass.